src/wallet/rpcwallet.cpp

```diff
@@ -1785,10 +1859,10 @@ UniValue gettransaction(const UniValue& params, bool fHelp)
     if (wtx.IsFromMe(filter))
         entry.pushKV("fee", ValueFromAmount(nFee));
 
-    WalletTxToJSON(wtx, entry);
+    WalletTxToJSON(wtx, entry, asOfHeight);
 
     UniValue details(UniValue::VARR);
-    ListTransactions(wtx, 0, false, details, filter);
+    ListTransactions(wtx, 0, false, details, filter, std::nullopt);
```

Comment:
Add a comment for why this passes nullopt.

Reply:
Should have been `asOfHeight`. Fixed.